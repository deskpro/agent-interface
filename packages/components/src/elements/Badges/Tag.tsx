import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/dp-badges.css";

export type TagProps = {
  color?: DPColor;
  size?: "small" | "medium";
};

const Tag: React.SFC<TagProps> = ({ children, color, size }) => (
  <span
    className={classNames("dp-Tag", {
      [`Tag--${color}`]: !!color,
      [`Tag--${size}`]: !!size
    })}
  >
    {children}
  </span>
);

export default Tag;
