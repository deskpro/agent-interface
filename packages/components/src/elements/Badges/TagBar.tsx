import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/components/dp-ListPane.css";

export type TagBar = {
  className?: string;
};

const TagBar: React.FC<TagBar> = ({ className, children }) => (
  <span className={classNames("dp-TagBar", className)}>{children}</span>
);

export default TagBar;
