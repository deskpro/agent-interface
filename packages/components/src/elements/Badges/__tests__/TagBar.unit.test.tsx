import * as React from "react";
import { render } from "react-testing-library";

import Tag from "../Tag";
import TagBar from "../TagBar";

describe("<TagBar />", () => {
  it("it should match its snapshot", () => {
    const { getByTestId } = render(
      <TagBar data-testid="tagbar">
        <Tag>First</Tag>
        <Tag>Second</Tag>
        <Tag>Third</Tag>
      </TagBar>
    );
    expect(getByTestId("tagbar")).toMatchSnapshot();
  });
});
