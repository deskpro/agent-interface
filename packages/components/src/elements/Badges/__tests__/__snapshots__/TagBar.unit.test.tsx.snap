// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagBar /> should match its snapshot 1`] = `
<span
  class="dp-TagBar dp-JustifyStart"
  data-testid="tagbar"
>
  <span
    class="dp-Tag"
  >
    First
  </span>
  <span
    class="dp-Tag"
  >
    Second
  </span>
  <span
    class="dp-Tag"
  >
    Third
  </span>
</span>
`;
